### ISSUE TITLE
`poetry run` could handle args better

### ISSUE THREAD

**Issue by rotu** [python-poetry/poetry#2741]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu 20.04
- **Poetry version**: 1.0.10
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: N/A

## Issue
When bad arguments are passed to `poetry run`, it is not immediately obvious what is wrong.

When no argument is provided, `poetry run` does not complain about the missing argument:
```
$poetry run

[FileNotFoundError]
[Errno 2] No such file or directory
```

Contrast with `poetry new`:
```
$poetry new
Not enough arguments (missing: "path").
```

Similarly, if given a flag, it might not be obvious that it is looking for an executable:
```
$poetry run --help

[FileNotFoundError]
[Errno 2] No such file or directory
```

**Comment by tomzx** [comment]
In the current implementation, it appears that when `poetry run` is called without any additional parameter, `run` is passed as the only argument to the `run` command. Effectively, it is the same as `poetry run run`, which explains why you are seeing a "No such file or directory error.

I'm not sure why `RunCommand` implements a custom argument parser. If I remove `RunCommand.__init__`, the code seems to behave as one would expect. Running the single test `test_run_passes_all_args` seems to indicate there is custom logic that is necessary as otherwise the CLI parser considers `poetry run python -V` as similar to `poetry -V` and thus gets interrupted during the `PRE_HANDLE` step of the command dispatcher.

What I suggested seems to be what @sdispater did @ 65ba952f2db1ce400a3153575db7c5b42f398885, which was then reverted back @ 8fb4f4524dbe8603e7fd7bba390b2a7e72eb3ae9, so it appears to be a current limitation with `clikit`.

**Comment by kirelagin** [comment]
> there is custom logic that is necessary as otherwise the CLI parser considers `poetry run python -V` as similar to `poetry -V`

A more-or-less standard approach is to say that, yes, this is how it is supposed to work and instead use the ` -- ` separator to stop parsing our arguments and pass everything else to the command. So your example would be `poetry run -- python -V`.

That will be a pretty major breaking UX change, so, as a compromise, maybe it would be better to say that everything after the first positional argument (command name) is arguments of the commands, but everything before it is ours. This would allow both `poetry run python -V` and `poetry run --help` to work.

I’d like to point out that `poetry run --help` not only not working, but also exiting with an extremely confusing error message is a pretty big deal.

**Comment by branchvincent** [comment]
These UX issues have been fixed on `master` and will be released with the upcoming `1.2`:

```console
$ poetry --version
Poetry (version 1.2.0a2)
$ poetry run

Not enough arguments (missing: "args")
$ poetry run --help

Description:
  Runs a command in the appropriate environment.
...
```

**Comment by west-github** [comment]
got Not enough arguments (missing: "args") when trying to run poetry run
